addons/point_of_sale/models/pos_order.py

```diff
@@ -47,17 +47,14 @@ def _order_fields(self, ui_order):
             'amount_total':  ui_order['amount_total'],
             'amount_tax':  ui_order['amount_tax'],
             'amount_return':  ui_order['amount_return'],
+            'company_id': self.env['pos.session'].browse(ui_order['pos_session_id']).company_id.id,
         }
 
-    def _payment_fields(self, ui_paymentline):
-        payment_date = ui_paymentline['name']
-        payment_date = fields.Date.context_today(self, fields.Datetime.from_string(payment_date))
+    @api.model
+    def _payment_fields(self, order, ui_paymentline):
         return {
-            'amount':       ui_paymentline['amount'] or 0.0,
-            'payment_date': payment_date,
-            'statement_id': ui_paymentline['statement_id'],
-            'payment_name': ui_paymentline.get('note', False),
-            'journal':      ui_paymentline['journal_id'],
+            'pos_order_id': order.id,
+            **ui_paymentline
```

Comment:
can you explicitly show the fields we are returning instead of '**ui_paymentline'